app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -323,6 +356,25 @@ private boolean hasFunds(List<PaymentMethod> clonedList) {
     return mergedList;
   }
 
+  private Integer mergeDisableReason(PaymentMethod appcMethod, PaymentMethod creditsMethod) {
+    Integer reason = null;
+
+    if (!creditsMethod.isEnabled()) {
+      if (creditsMethod.getDisabledReason() != null) {
+        reason = creditsMethod.getDisabledReason();
+      } else {
+        reason = appcMethod.getDisabledReason();
+      }
+    } else if (!appcMethod.isEnabled()) {
+      if (appcMethod.getDisabledReason() != null) {
+        reason = appcMethod.getDisabledReason();
+      } else {
+        reason = creditsMethod.getDisabledReason();
+      }
+    }
+    return reason;
+  }
```

Comment:
We should keep the method of the reason in case that in the future we want to update the MergeAppCoins view to update on resume. Plus I'm seeing a lot of duplicated code, somethings can be extracted